src/leveldb/db/memtable.cc

```diff
@@ -101,7 +101,7 @@ void MemTable::Add(SequenceNumber s, ValueType type,
   p += 8;
   p = EncodeVarint32(p, val_size);
   memcpy(p, value.data(), val_size);
-  assert((p + val_size) - buf == encoded_len);
+  assert((buf < (p + val_size)) && (abs((p + val_size) - buf) == encoded_len));
```

Comment:
This is an interesting solution, but I don't think we need to change this warning.

leveldb a library included with the rest of the source, but it is not something that we would want to change.

The reason why this is a warning is because differences between pointers results in a signed integer and that is being compared to an unsigned integer.  The size of the data written is certainly less than the maximum signed integer size, so there shouldn't be a case where this assert tests positively, when it should fail.